holoviews/core/data/narwhals.py

```diff
@@ -0,0 +1,620 @@
+
+        dropped = []
+        columns = list(df.collect_schema())
+        for vd in vdims:
+            if vd not in columns:
+                dropped.append(vd)
+        return df, dropped
+
+    @classmethod
+    def unpack_scalar(cls, dataset, data):
+        """Given a dataset object and data in the appropriate format for
+        the interface, return a simple scalar.
+
+        """
+        is_scalar, is_lazy = cls._is_scalar_size(data)
+        if is_scalar:
+            return (data.collect() if is_lazy else data).item()
+        return data
+
+    @classmethod
+    def _is_scalar_size(cls, data) -> tuple[bool, bool]:
+        is_lazy = isinstance(data, nw.LazyFrame)
+        cols = data.collect_schema()
+        ncols = len(cols)
+        if ncols != 1:
+            return False, is_lazy
+        nrows = data.select(nw.col(next(iter(cols))).len())
+        nrows = (nrows.collect() if is_lazy else nrows).item()
+        return nrows == 1, is_lazy
+
+    @classmethod
+    def mask(cls, dataset, mask, mask_value=None):
+        data = dataset.data
+        if not dataset.vdims:
+            return data
+
+        if data.implementation != nw.Implementation.POLARS:
+            # polars will convert None to Null
+            mask_value = np.nan
+
+        cols = [vd.name for vd in dataset.vdims]
+        mask_series = nw.new_series(
+            name="__mask__",
+            values=mask,
+            backend=_EAGER_TYPE.get(data.implementation, data.implementation),
+        )
+        return (
+            data.with_columns(mask_series)
+            .with_columns(
+                [
+                    nw.when(nw.col("__mask__"))
+                    .then(nw.col(col))
+                    .otherwise(mask_value)
+                    .alias(col)
+                    for col in cols
+                ]
+            )
+            .drop("__mask__")
+        )
+
+    @classmethod
+    def redim(cls, dataset, dimensions):
+        column_renames = {k: v.name for k, v in dimensions.items()}
+        return dataset.data.rename(column_renames)
+
+    @classmethod
+    def sort(cls, dataset, by=None, reverse=False):
+        if by is None:
+            by = []
+        cols = [dataset.get_dimension(d, strict=True).name for d in by]
+        return dataset.data.sort(by=cols, descending=reverse)
+
+    @classmethod
+    def select(cls, dataset, selection_mask=None, **selection):
+        df = dataset.data
+        if selection_mask is None:
+            column_sel = {k: v for k, v in selection.items()}
+            if column_sel:
+                selection_mask = cls.select_mask(dataset, column_sel)
+
+        if selection_mask is not None:
+            if isinstance(selection_mask, np.ndarray):
+                # Boolean ndarray does not work, so we convert it to list
+                # If the dtype is not boolean, we let narwhals error in filter
+                selection_mask = selection_mask.tolist()
```

Comment:
Hmm, that sucks.

Reply:
That seems more elegant, thanks @FBruzzesi.

And you are always welcome to review the code :+1: